openpype/hosts/hiero/api/plugin.py

```diff
@@ -705,9 +705,10 @@ def __init__(self, cls, track_item, **kwargs):
         self._create_parents()
 
     def convert(self):
-
         # solve track item data and add them to tag data
-        self._convert_to_tag_data()
+        tag_hierarchy_data = self._convert_to_tag_data()
+
+        self.tag_data.update(tag_hierarchy_data)
```

Comment:
just a naive note about `self.tag_data` - since it is updated with the content of `tag_hierarchy_data`, shouldn't it be used late on instead of it? Also, this whole method is returning `tag_hierarchy_data` but the instance property `tag_data` already has all of it but is different?